### PR TITLE
added homework_21_12 and additional task

### PR DESCRIPTION
**Pull request by Viktoriyadneprovskaya**

### CODE REVIEW

File: homework_21_12_dop/src/StringMethods.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import java.util.Arrays;
+
+
+public class StringMethods {
+    String str = "";
```
Comment (YeaaaaH) [inline homework_21_12_dop/src/StringMethods.java]: Why do you declare it?

File: homework_21_12_dop/src/StringMethods.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    StringMethods(String str) {
```
Comment (YeaaaaH) [inline homework_21_12_dop/src/StringMethods.java]: You don't need this constructor

File: homework_21_12_dop/src/StringMethods.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        while (i > j) {
+            if (str3.charAt(i) != str3.charAt(j)) {
+                return false;
+            }
+        i--;
+        j++;
+        }
+        return true;
+    }
+    public void validate(boolean a){
```
Comment (YeaaaaH) [inline homework_21_12_dop/src/StringMethods.java:36]: do you really need this method?

File: homework_21_12_dop/src/StringMethods.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
+import java.util.Arrays;
+
+
+public class StringMethods {
+    String str = "";
+    StringMethods(String str) {
+        this.str = str;
+    }
+
+//    public int povtor(String str) {
```
Comment (YeaaaaH) [inline homework_21_12_dop/src/StringMethods.java]: naming...

File: homework_21_12_dop/src/StringMethods.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+//                        maxCounter = counter;
+//                    }
+//                }
+//            }
+//            counter=0;
+//        }
+//        return maxCounter;
+//    }
+
+    public void delChar(String str) {//не работает
```
Comment (YeaaaaH) [inline homework_21_12_dop/src/StringMethods.java]: you can make these methods static

**Review by YeaaaaH — CHANGES_REQUESTED** [review]
need to fix

File: homework_21_12_dop/src/StringMethods.java (hunk 1 above)
Comment (Viktoriyadneprovskaya) [reply homework_21_12_dop/src/StringMethods.java]: deleted

File: homework_21_12_dop/src/StringMethods.java (hunk 2 above)
Comment (Viktoriyadneprovskaya) [reply homework_21_12_dop/src/StringMethods.java]: deleted

File: homework_21_12_dop/src/StringMethods.java (hunk 3 above)
Comment (Viktoriyadneprovskaya) [reply homework_21_12_dop/src/StringMethods.java:36]: this method is used to make output more readable and understandable
I mean to output string massage, but not a false or true

File: homework_21_12_dop/src/StringMethods.java (hunk 4 above)
Comment (Viktoriyadneprovskaya) [reply homework_21_12_dop/src/StringMethods.java]: corrected

File: homework_21_12_dop/src/StringMethods.java (hunk 5 above)
Comment (Viktoriyadneprovskaya) [reply homework_21_12_dop/src/StringMethods.java]: done

**Pull request merged**